pyvrp/cpp/search/LocalSearch.cpp

```diff
@@ -66,19 +66,10 @@ Solution LocalSearch::search(Solution &solution,
                 }
             }
 
-            if (step > 0)  // empty moves are not tested initially to avoid
-            {              // using too many routes.
-                auto pred = [](auto const &route) { return route.empty(); };
-                auto empty = std::find_if(routes.begin(), routes.end(), pred);
-
-                if (empty == routes.end())
-                    continue;
-
-                if (U->route)  // try inserting U into the empty route.
-                    applyNodeOps(U, empty->depot, costEvaluator);
-                else  // U is not in the solution, so again try inserting.
-                    maybeInsert(U, empty->depot, costEvaluator);
-            }
+            // Empty route moves are not tested in the first iteration to avoid
+            // increasing the fleet size too much.
+            if (step > 0 && applyNodeOpsWithEmptyRoutes(U, costEvaluator))
+                continue;
```

Comment:
I like the simplicity but this version evaluates the operators for all empty routes (also in the homogeneous case) which I'd really like to avoid (e.g. in EURO-NeurIPS we had `num_vehicles = num_clients`). Please see what I've implemented now, it largely keeps the current implementation and only adds a few lines around it, and it checks all vehicle types, now that it's no longer a seperate function it is also more inline with the other loops.